### DISCUSSION TITLE
Bulk .upsert() on table with composite primary key

### DISCUSSION THREAD

**Discussion by pranavbadami** [supabase/supabase#3127] [Questions] [answered]
Hey folks!

I'm currently considering some DB changes where I would be changing one of my tables to have a composite primary key. Currently, as part of my ETL process, I am using [upsert()](https://supabase.io/docs/reference/javascript/upsert) to bulk insert/update rows into this table. This works like a charm with a single-column primary key, but will this continue to work as expected with a composite key?

**Comment by steve-chavez** [comment, marked as answer]
Hey @pranavbadami,

>  This works like a charm with a single-column primary key, but will this continue to work as expected with a composite key?

Yes, it will. This is [tested to work on PostgREST](https://github.com/PostgREST/postgrest/blob/0ba133a0bd1e41efa1c914e50d80b3270d52db29/test/Feature/UpsertSpec.hs#L35-L46) and thus it should work transparently on postgrest-js/supabase-js.
